### PR TITLE
Updated electron-related dependencies

### PR DESCRIPTION
**Pull request by ruyadorno**
- Update **electron** and its related deps to its latests versions :zap: 
- Fixed API breaking changes :hammer: 

@stefanbuck

### CODE REVIEW

**Comment by stefanbuck** [comment]
That's so wonderful. This was in my mind since a couple of weeks. Thanks for taking care of it. If you need help, shout.

**Comment by ruyadorno** [comment]
hmmmm, something is very odd with the CI...

```
Xlib:  extension "RANDR" missing on display ":99.0".
```

can you please take a look at travis and see if it rings a bell? my tests passed locally

File: src/browser/application.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import app from 'app';
```
Comment (sindresorhus) [inline src/browser/application.js]: For forward compatibility, you should update all electron modules to be off the `require('electron')` import.

See: https://github.com/atom/electron/releases/tag/v0.35.0

Comment (ruyadorno) [inline src/browser/application.js]: :+1: good call!

I was going for the minimal effort to get things running and missed this :blush:  I'll update the PR to also take care of the other electron imports.

thanks :muscle:

**Comment by addyosmani** [comment]
Unfortunately, it still looks like this is failing on Travis even with the Node 5 updates made. I wonder if the RANDR issues we're seeing are related to https://ask.mozilla.org/question/1426/xlib-extension-randr-missing-on-display-990/.

**Comment by stefanbuck** [comment]
That's not the issue. Travis is failing because @ruyadorno changed the require statements to import for the browser process files in 8bede22663dc65f2eedaaea77c5b3121aaed1432. 

The ES6 import statements are not supported in the browser process, just in the render process (window), because the browser process behaves like a normal node, by that import are not working there. Maybe it's time to use babel for this area as well to avoid any further confusion.

**Comment by ruyadorno** [comment]
:+1: I'll revert that change for now

**Comment by ruyadorno** [comment]
hmmm, that wasn't really the issue :disappointed:

**Comment by stefanbuck** [comment]
Mhhhh, strange it's working on my branch https://travis-ci.org/stefanbuck/yeoman-app/builds/101911358. Note this does not include your work (it's outdated), but it was working without modifying `.travis.yml`.

**Comment by ruyadorno** [comment]
yeah, that last commit did it! now tests are failing on jasmine, I'm extending lunch time a little bit to try to figure it out :smile:

**Comment by ruyadorno** [comment]
:confetti_ball: it works!

@stefanbuck if everything looks good to you I'll squash it and let's get it merged :shipit:

**Comment by stefanbuck** [comment]
Great job @ruyadorno. Only one thing is left, `npm run build` is still not working, but I'm on it.

**Comment by ruyadorno** [comment]
@stefanbuck ok, just update the PR with the squashed commits

**Comment by ruyadorno** [comment]
maybe we can merge this already and tackle the build fix in a separate PR, as far as I could tell it wasn't working for me anyways as I reported on #96

**Comment by stefanbuck** [comment]
I would prefer to tackle this within this PR, because the build was working for me as expected with the current master. Is there any reason to rush with this one?

**Comment by ruyadorno** [comment]
:+1: sure! I'm just waiting for this PR to land to resume work on implementing the renderer components tests - that said, no, there's no rush :walking: 

just let me know once you're done and I can squash the commit into your changes :wink:

**Comment by ruyadorno** [comment]
@stefanbuck I just updated the PR with your changes, can you checkout the branch and test it in your machine? I still couldn't get the built app to run on mine :disappointed:

**Comment by stefanbuck** [comment]
Again, it's working fine for me. Can anyone else confirm that `npm run build` produces a corrupt `Yeoman.app` file?

**Comment by ruyadorno** [comment]
ok, nevermind, something was borked in my working repo - tried your steps in a separate clean folder and it works :blush: 

we're good to go :shipit:

**Comment by stefanbuck** [comment]
Awesome, let's merge it :smile:

**Comment by ruyadorno** [comment]
:confetti_ball:

**Pull request merged**